### PR TITLE
Checking whether everything still works

### PR DESCRIPTION
**Pull request by 2bigO**
- `/api/jobs`
- getting `/api/new` to work again
- agendaReady for bootstrapping
- test for `/api/new`

### CODE REVIEW

**Comment by 2bigO** [comment]
This is also a first stab at tests for #4

File: src/index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const agenda = new Agenda({
   }
 });
 
-agenda.on('ready', () => {
+const agendaReady = new Promise(resolve => agenda.on('ready', () => {
```
Comment (simison) [inline src/index.js:33]: 👍 👍 👍

File: src/index.js [hunk 2]
```diff
@@ -33,19 +33,35 @@ agenda.on('ready', () => {
...
+    .then(job => Promise.all([agendaReady, job]))
+    .then(([jobs, job]) => defineJob(agenda, jobs, job))
     .catch(err => {
       ctx.status = 400;
       return err.message;
     });
   await next();
 });
 
+router.get('/api/jobs', async (ctx, next) => {
```
Comment (simison) [inline src/index.js]: 👏 👏

File: test.js [hunk 3]
```diff
@@ -25,3 +20,11 @@ test.serial('POST /api/new fails without content', async t => {
 
   t.is(res.status, 400);
 });
+
+test.serial('POST /api/new succeeds when a job is specified', async t => {
+  const res = await request('http://localhost:4041')
```
Comment (simison) [inline test.js]: BTW I was hoping to do `request(bootstrapApp())` or `request(app)` but somehow I was getting "app.address is not a function" errors and went on with using URL directly here. 🤷‍♂️

File: src/index.js (hunk 1 above)
Comment (simison) [reply src/index.js:33]: Note that this might go in at some point: https://github.com/agenda/agenda/pull/557

**Comment by simison** [comment]
Loving the title 😆 

Looks good, just we might need to refactor once agenda uses promises.

File: src/index.js (hunk 2 above)
Comment (2bigO) [reply src/index.js]: 👍 👍

File: src/index.js (hunk 1 above)
Comment (2bigO) [reply src/index.js:33]: Yeah, sure, we'd have to refactor when that comes in

**Comment by simison** [comment]
I'll give this a test spin little later unless anyone else is faster to get to it 👍

File: test.js (hunk 3 above)
Comment (2bigO) [reply test.js]: I'm not familiar with `supertest`, so I'm not really getting what you mean 😆

Comment (OmgImAlexis) [reply test.js]: @simison I'll fix that up later.

File: src/job.js [hunk 4]
```diff
@@ -1,9 +1,35 @@
...
+
+const getAssertFunction = (assertOnCount, errorOnName) => async (job, jobs) => countJobByName(job.name, jobs)
+  .then(count => {
+    if (!assertOnCount(count)) {
+      throw new Error(errorOnName(job.name));
+    }
+  });
+
+const jobAssertions = {
+  allreadyExists: getAssertFunction(count => count > 0, name => `Did not find a job named "${name}"`),
```
Comment (OmgImAlexis) [inline src/job.js]: `alreadyExists`

File: test.js [hunk 5]
```diff
@@ -1,27 +1,30 @@
...
 test.before.cb(t => {
-  bootstrapApp();
-  agenda.on('ready', () => {
-    t.end();
-  });
+  bootstrapApp().then(() => t.end());
 });
 
-test.serial('POST /api/new fails without content', async t => {
+test.serial('POST /api/job fails without content', async t => {
```
Comment (OmgImAlexis) [inline test.js:16]: `/api/job` or `/api/jobs`

Comment (OmgImAlexis) [reply test.js:16]: You have it different in title vs test.

Comment (keyvan-m-sadeghi) [reply test.js]: yeah got your message a bit late

Comment (2bigO) [reply test.js:16]: Changed back to singular

**Review by OmgImAlexis — APPROVED** [review]

**Review by simison — APPROVED** [review]
Did very cursory testing and read the code — LGTM, it does return jobs from agenda db at least. :-)

Feel free to merge!

**Comment by simison** [comment]
@keyvan-m-sadeghi recommend squash+merge next time to avoid too many commit messages ;-) but not a biggie. 👍

**Pull request merged**